prime-router/src/test/kotlin/fhirengine/engine/FhirConverterTests.kt

```diff
@@ -270,6 +271,36 @@ class FhirConverterTests {
         }
     }
 
+    @Test
+    fun `test getContentFromHL7 alternate profile`() {
+        val testProfile = HL7Reader.Companion.MessageProfile("ORU", "PHLabReport-NoAck")
+
+        mockkClass(HL7Reader::class)
+        mockkObject(HL7Reader.Companion)
+        every { HL7Reader.Companion.messageProfileMap.get(testProfile) } returns "./metadata/test_fhir_mapping"
+
+        val actionLogger = spyk(ActionLogger())
+        val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.process) as FHIRConverter)
+        val message = spyk(
+            FhirConvertQueueMessage(
+                UUID.randomUUID(),
+                BLOB_URL,
+                "test",
+                BLOB_SUB_FOLDER_NAME,
+                topic = Topic.FULL_ELR
+            )
+        )
+
+        every { message.downloadContent() }
+            .returns(validHl7)
+
+        val result = engine.getContentFromHL7(message, actionLogger)
+
+        // the test fhir mappings produce far fewer entries than the production ones
+        assertThat(result).isNotEmpty()
+        assertThat(result[0].entry.size).isEqualTo(5)
```

Comment:
Open to suggestions to make this test less brittle. This currently relies on the test mappings producing a specific number of entries.